faq.html

```diff
@@ -161,6 +161,87 @@ <h2 id="outputformat">The output format of ParticleFile</h2>
+  <p></p>
+  Field chunking is introduced in Parcels > 2.0.0. It manages the access and data transfer between your data / model (i.e. your NetCDF files), the memory, and the kernels, in a way so that only the parts of the data ends up in memory which are directly needed by some particles. This technique is commonly referred to as <i>dynamic loading</i>. In Python, with its numeric backend of Numpy and SciPy, the <a href="https://dask.org/">Dask</a> module implements this data access technique, which is called <i>chunking</i> or <i>lazy loading</i> within Dask. Chunking for field data is activated by <i>default</i> in Parcels > 2.0.0.
+  <p></p>
+  As an example, let us assume we load a field set as such:
+  <p></p>
+  <code class="language-python">fieldset = FieldSet.from_netcdf(files, variables, dimensions)</code>
+  <p></p>
+  then, this is equivalent to
+  <p></p>
+  <code class="language-python">fieldset = FieldSet.from_netcdf(files, variables, dimensions, deferred_load=True, field_chunksize='auto')</code>
+  <p></p>
+  Hence, if you are using an older version of Parcels (<= 2.0.0), or you have accidentally switched of chunking  via <code class="language-python">fieldset = FieldSet.from_netcdf(files, variables, field_chunksize=False)</code>, then you may exceed your available memory. Try out a newer version of Parcels (>= 2.1.3) and <i>activate</i> field chunking.
+
+  <p></p>
+  <h4>Advanced control over chunking and memory - configure Dask</h4>
+  <p></p>
+  In the case that you have field chunking running, your job submission files are all set correctly, but your fields are still too big, then your fields must be either many (in terms of <i>individual</i> fields), or they are complex (i.e. many diverse, hydrological properties), or they are in dense (i.e. <i>high-resolution</i>) 4D grids (i.e. using time, depth, latitude and longitude).
+  In this case, you may need to tune your environment a bit more directly to your application or scenario.
+  <ol>
+    <li> <b>Attune your chunk size configuration manually</b>: Instead of leaving the chunking up to the black-box that is Dask, you can define your chunks yourself. Say you have simulation data of 30 timesteps on a grid with depth=150, lat=2100, long=1800 (just an example), then you may wanna define the chunks yourself.
+      <ul>
+        <li>use the parameter <i>field_chunksize</i> of the FieldSet class as a tuple: here, the order is <code class="language-python">(time steps, depth size, latitude size, longitude size)</code>, so e.g. <code class="language-python">fieldset = FieldSet.from_netcdf(files, variables, dimensions, field_chunksize=(1,8,128,128))</code></li>
```

Comment:
Would it not be simpler/safer to suggest users to always provide a dictionary? I.e. to remove this suggestion of a tuple?

Reply:
well, not necessarily - the dictionary was actually more difficult to set up than the tuple. That is because the chunking is a **Dask** feature, not essentially an **xarray** feature, which means the `dask.array.core.Array.rechunk()` method itself operates on numeric indices and tuples, not on named variables and dictionaries. That is why I had to provide a complicated way to translate dictionaries in the end. So, long story short: if you know your indices, it's good to use the tuple. Only if you're not sure with your index order, use the dictionary. 

And at least I wanted to list all possibilities.

Reply:
Ah ok, I understand. Then leave as is